### PR TITLE
MM-31187 Enable broadcast to private channels.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Enables selection of private channels in the broadcast option. 

Also fixes bug where users could possibly select channels they are not a part of by guessing an ID. 

If another member of the playbook is not a member of the channel selected, they will see "Unknown Channel" in the dropdown. They will still be able to change the values of other fields, or select a channel they are a member of. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31187

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]
Nice!

File: webapp/src/websocket_events.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function handleReconnect(getState: GetStateFunc, dispatch: Dispatch) {
 }
 
 export function handleWebsocketIncidentUpdated(getState: GetStateFunc, dispatch: Dispatch) {
-    return (msg: WebSocketMessage): void => {
+    return (msg: WebSocketMessage<{payload: string}>): void => {
```
Comment (cpoile) [inline webapp/src/websocket_events.ts:37]: Just curious, why are we adding this (and the ones below)?

**Review by cpoile — APPROVED** [review]
Thanks! Just a non-blocking question.

File: webapp/src/websocket_events.ts (hunk 1 above)
Comment (crspeller) [reply webapp/src/websocket_events.ts:37]: There was a breaking change in the type exported by mattermost-redux. I updated mattermost-redux, so I needed to fix the types here.

**Review by prapti — APPROVED** [review]

**Pull request merged**